module3/src/MyList.java

```diff
@@ -0,0 +1,697 @@
+import java.util.*;
+
+/**
+ * Created by yurii.pyvovarenko on 11/28/2021.
+ */
+public class MyList<T> implements List<T>, Iterable<T> {
+    private static final Object[] DEFAULTCAPACITY_EMPTY_ELEMENTDATA = {};
+    private static final int DEFAULT_CAPACITY = 10;
```

Comment:
Your default capacity is 10 but your `DEFAULTCAPACITY_EMPTY_ELEMENTDATA` has size 0 by default and it will be always extended within the first call of  `add` operation. Would be better to have array with default size 10

Reply:
reworked constructor and dropped usage of `DEFAULTCAPACITY_EMPTY_ELEMENTDATA`
```
    public MyList() {
        this.elementData = new Object[DEFAULT_CAPACITY];
    }
```